src/common/filesystem.cpp

```diff
@@ -0,0 +1,21 @@
+#include "filesystem.h"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
```

Comment:
This would fail on Windows, wouldn't it? Assuming that we are not going to use pipes on Windows, can you surround that with appropriate #ifdefs and make sure things work with the compilation of these parts turned off?

Reply:
We seem to be using "#ifdef _MSC_VER" for that purpose. @frankseide is that the preferred way?

Reply:
Resolving as I fix things.